File: Pk5Mining.Server/Models/Departments/Department.cs
using Pk5Mining.Server.Models.Subsidiaries;
using System.ComponentModel.DataAnnotations.Schema;

namespace Pk5Mining.Server.Models.Departments
{
    public class Department
    {
        public long Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public long SubsidiaryId { get; set; }
        public bool IsActive { get; set; } = true;
        public DateTime DT_Created { get; set; }
        public DateTime? DT_Modified { get; set; }

        [ForeignKey("SubsidiaryId")]
        public virtual Subsidiary? Subsidiary { get; set; }
    }
}

File: PK5Mining.RecruitmentPortal/src/app/interfaces/department.ts
import { PaginationInfo } from ".";
import { Subsidiary } from "./subsidiary";

export type Department = {
  id: string;
  name: string;
  description?: string;
  subsidiaryId?: string;
  subsidiary?: Subsidiary | null;
  isActive?: boolean;
  status: "Active" | "Inactive";
  dT_Created: string;
  dT_Modified: string;
  dT_Updated: string,
};

export type DepartmentResponsePayload = {
  data: Department[];
} & PaginationInfo;

export type CreateDepartmentPayload = Omit<Department, "id" | "dT_Created" | "dT_Modified">;

export type UpdateDepartmentPayload = Partial<CreateDepartmentPayload>;

export type DepartmentErrors = {
  [K in keyof CreateDepartmentPayload]?: string;
};

export type DepartmentsQuery = {
  pageNumber: number;
  pageSize: number;
  name?: string;
  search?: string;
  isActive?: boolean | string;
};